File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.swift
import Foundation

func solution(_ number:Int, _ limit:Int, _ power:Int) -> Int {
    var result: [Int] = Array(repeating: 0, count: number + 1)
    
    for num in 1...number {
        var sqrtOfNum = Int(sqrt(Double(num)))
        var cnt = 0
        
        for i in 1...sqrtOfNum {
            if num % i == 0 {
                ((num / i) == i) ? (cnt += 1) : (cnt += 2)
            }
        }
        result[num] = cnt
    }
    
    return result
        .map { ($0 > limit) ? power : $0  }
        .reduce(0, +)
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<int> v;

int getPrimeCount(const int& num) {
    int count = 0;

    for (int i = 1; i * i <= num; i++) {
        if (num % i == 0) {
            (i * i == num) ? count += 1 : count += 2;
        }
    }

    return count;
}

int solution(int number, int limit, int power) {
    int answer = 0;

    for (int i = 1; i <= number; i++) {
        v.push_back(getPrimeCount(i));
    }

    for (int i = 0; i < v.size(); i++) {
        (v[i] <= limit) ? answer += v[i] : answer += power;
    }

    return answer;
}